lib/arm/crc32_impl.h

```diff
@@ -151,12 +152,66 @@ crc32_pmull_aligned(u32 remainder, const uint8x16_t *p, size_t nr_segs)
 #include "../crc32_vec_template.h"
 #endif /* PMULL implementation */
 
+#undef DISPATCH_INSTR
+#if (defined(__ARM_FEATURE_CRC32) ||	\
+    (ARM_CPU_FEATURES_ENABLED &&	\
+     COMPILER_SUPPORTS_CRC32_TARGET_INTRINSICS)) && \
+     /* not yet tested on big endian, probably needs changes to work there */ \
+    (defined(__BYTE_ORDER__) && __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__)
+#  define FUNCNAME		crc32_instr
+#  define FUNCNAME_ALIGNED	crc32_instr_aligned
+#  define FUNCNAME_SLICE1	crc32_instr_slice1
+#  ifdef __ARM_FEATURE_CRC32
+#    define ATTRIBUTES
+#    define DEFAULT_IMPL	crc32_instr
```

Comment:
`DEFAULT_IMPL` may have already been defined.  How are you trying to order the default implementations?  I.e. if someone compiles with both pmull and crc32 instructions enabled, which implementation do you intend to use?  How does performance compare?  Check `lib/x86/crc32_impl.h` for how this is handled for x86.  E.g. if the crc32 instructions are meant to be first priority, you have to define it first, and then only define the pmull implementation if `!defined(DEFAULT_IMPL)` already.